Title: AntreaIPAMController panics on StatefulSet delete when informer delivers tombstone object

**Description:**

`AntreaIPAMController.enqueueStatefulSetDeleteEvent` in `pkg/controller/ipam/antrea_ipam_controller.go` performs an unchecked type assertion on the `obj` parameter:

```go
ss := obj.(*appsv1.StatefulSet)
```

When the informer's watch connection is interrupted and later re-established (e.g. API server restart, network partition, or controller restart), Kubernetes may deliver delete events as `cache.DeletedFinalStateUnknown` tombstone objects instead of `*appsv1.StatefulSet`.

In this case, the type assertion panics, causing the `antrea-controller` process to crash.

This pattern is already handled correctly in other parts of the codebase (e.g. Node delete handlers), but is missing here.

---

**Steps to Reproduce:**

1. Deploy Antrea with AntreaIPAM enabled and at least one StatefulSet using an IPPool
2. Interrupt the controller’s watch connection (e.g. restart API server or controller)
3. Delete the StatefulSet during the interruption
4. On watch reconnection, the delete event is delivered as a tombstone
5. The controller panics

---

**Expected behavior:**

The delete event should be handled gracefully by extracting the StatefulSet from the tombstone and enqueueing it.

---

**Actual behavior:**

The controller panics:

```
panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.StatefulSet
```

---

**Root Cause:**

File: `pkg/controller/ipam/antrea_ipam_controller.go`

The delete handler does not handle `cache.DeletedFinalStateUnknown`, which is a standard Kubernetes mechanism for delivering missed delete events.

---

**Impact:**

High. This causes the `antrea-controller` to crash during common operational scenarios (e.g. API server restart), disrupting IPAM and potentially leaving IP allocations unreleased.

---

**Proposed Fix:**

Add tombstone handling similar to other controllers:

```go
ss, ok := obj.(*appsv1.StatefulSet)
if !ok {
    tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
    if !ok {
        klog.ErrorS(nil, "Unexpected object in StatefulSet delete handler", "obj", obj)
        return
    }
    ss, ok = tombstone.Obj.(*appsv1.StatefulSet)
    if !ok {
        klog.ErrorS(nil, "Tombstone contains non-StatefulSet object", "obj", tombstone.Obj)
        return
    }
}
```

---

**Additional Work:**

* Add a unit test in `antrea_ipam_controller_test.go` to verify tombstone handling

--- a/pkg/controller/ipam/antrea_ipam_controller.go
+++ b/pkg/controller/ipam/antrea_ipam_controller.go
@@ -163,7 +163,21 @@ func (c *AntreaIPAMController) enqueueStatefulSetCreateEvent(obj interface{}) {
 
 // Enqueue the StatefulSet delete notification to be processed by the worker
 func (c *AntreaIPAMController) enqueueStatefulSetDeleteEvent(obj interface{}) {
-	ss := obj.(*appsv1.StatefulSet)
+	ss, ok := obj.(*appsv1.StatefulSet)
+	if !ok {
+		// When the informer's watch connection is interrupted and re-established,
+		// delete events are delivered as cache.DeletedFinalStateUnknown tombstones.
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(nil, "Error decoding object when deleting StatefulSet, invalid type", "object", obj)
+			return
+		}
+		ss, ok = tombstone.Obj.(*appsv1.StatefulSet)
+		if !ok {
+			klog.ErrorS(nil, "Error decoding object tombstone when deleting StatefulSet, invalid type", "object", tombstone.Obj)
+			return
+		}
+	}
 	klog.V(2).InfoS("Delete notification", "Namespace", ss.Namespace, "StatefulSet", ss.Name)
 
 	key := k8s.NamespacedName(ss.Namespace, ss.Name)

--- a/pkg/controller/ipam/antrea_ipam_controller_test.go
+++ b/pkg/controller/ipam/antrea_ipam_controller_test.go
@@ -39,6 +39,7 @@ import (
 	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions"
 	listers "antrea.io/antrea/pkg/client/listers/crd/v1beta1"
 	annotation "antrea.io/antrea/pkg/ipam"
+	"antrea.io/antrea/pkg/util/k8s"
 )
 
 type fakeAntreaIPAMController struct {
@@ -438,4 +439,21 @@ func BenchmarkCleanUpStaleIPAddresses(b *testing.B) {
 			}
 		})
 	}
+}
+
+func TestEnqueueStatefulSetDeleteEvent_Tombstone(t *testing.T) {
+	namespace, pool, statefulSet := initTestObjects(false, true, 3)
+	controller := newFakeAntreaIPAMController(pool, namespace, statefulSet)
+
+	expectedKey := k8s.NamespacedName(statefulSet.Namespace, statefulSet.Name)
+
+	// Deliver the delete as a tombstone, as the informer does after a watch reconnect.
+	controller.enqueueStatefulSetDeleteEvent(cache.DeletedFinalStateUnknown{
+		Key: expectedKey,
+		Obj: statefulSet,
+	})
+
+	require.Equal(t, 1, controller.statefulSetQueue.Len())
+	key, _ := controller.statefulSetQueue.Get()
+	assert.Equal(t, expectedKey, key)
 }

